R/longData.R

```diff
@@ -185,27 +185,39 @@ longDataConstructor <- R6::R6Class(
         #' A `data.frame`.
         get_data = function(obj = NULL, nmar.rm = FALSE, na.rm = FALSE, idmap = FALSE) {
 
-            if (is.null(obj)) return(self$data)
+            if (is.null(obj)) {
+                if (nmar.rm == FALSE & na.rm == FALSE) {
+                    return(self$data)
+                } else {
+                    ids <- self$ids
+                    list_flag <- FALSE
+                    orig_data_flag <- TRUE
+                }
 
-            if (! any(c("imputation_df", "character") %in% class(obj))) {
-                stop("Object must be an imputation_df or a character vector")
-            }
+            } else {
```

Comment:
Plus, now basically the main change is that even if `obj` is `NULL` we remove either mnar or na values (or both). There is no difference in the behavior if `obj` is  `NULL` or if it is not `NULL` (the only thing that changes is how ids are extracted..). Correct?

Reply:
I'm not sure I fully understand your question but will try to talk around it,

So we check for NULL still because if  `obj = NULL`, `nmar.rm  = FALSE`  and `na.rm = FALSE` then we can quick exit by returning the full dataset without any other processing. 

If `obj = NULL` but `nmar.rm  = TRUE`  or `na.rm = TRUE` then we derive which observations need to be removed as usual but suppress the code that changes the subject ids (this is required when bootstrapping to ensure unique patient ids for repeated patients.